### DISCUSSION TITLE
How to use spaces, '/' , '\' 'Single quote', "Double quote" inside a --package-parameters in Choco install command

### DISCUSSION THREAD

**Discussion by abhishekm1359** [chocolatey-community/chocolatey-packages#2514] [Q&A] [answered]
I am trying to install a MSI file by passing a parameters like below

choco install Msiname --version '1.0.0' --package-parameters="'/filelocation:D:\Artifacts\MSIS\Server.msi' '/arguments:USERNAME=".\administrator"' " (this works fine),
 but i am getting error when passing parameter with spaces
choco install Msiname--version '1.0.0' --package-parameters="'/filelocation:D:\Artifacts\MSIS\Server.msi' '/arguments:CERTIFICATE="c:\cert test\new.corp.pfx"' "(throws error in "cert test" because of space) 
What if i want to use quotes inside a quotes like " " " (double quote inside a double quotes) or ' ' ' (single quote inside a single quotes)   

How can i solve this.

**Comment by pauby** [comment, marked as answer]
This is not a Chocolatey CLI issue. This is a shell issue. If you're using the PowerShell shell, then you need to properly escape those characters.

See [scripting best practices](https://docs.chocolatey.org/en-us/choco/commands/#scripting--integration---best-practices--style-guide).
